libtransact/src/scheduler/serial/core.rs

```diff
@@ -324,6 +333,14 @@ impl SchedulerCore {
                             ));
                         }
                         ExecutionTaskCompletionNotification::Invalid(_context_id, result) => {
+                            if &result.transaction_id != current_txn_id {
+                                warn!(
+                                    "received execution result for a transaction ({}) that is \
+                                     not the current transaction ({})",
+                                    result.transaction_id, current_txn_id
+                                );
+                                continue;
```

Comment:
Shouldn't this still invalidate the whole batch, instead of continuing?  If it's not the current transaction, is it even a transaction in the current batch?

Reply:
We could check if the transaction is in the current batch, but just like with a valid transaction, this indicates that something is wrong with either the scheduler or the executor, so we should probably either panic or ignore in those cases respectively?

Reply:
This probably should also be sent to the forthcoming Error Callback.